src/libra_py/auxiliary_functions.py

```diff
@@ -0,0 +1,547 @@
+                [ [9,10], "alp" ] --> [5,6]
+                [ [9,10], "bet" ] --> [-15,-16]
+         In general form, the reindexing take the form of:
+                [ [9,10], "alp" ] --> [ index from 1 of 9 in ks_orbital_indicies, index from 1 of 10 in ks_orbital_indicies  ]
+                [ [9,10], "bet" ] --> [ - ( index from 1 of 9 in ks_orbital_indicies + len(ks_orbital_indicies) ), - ( index from 1 of 10 in ks_orbital_indicies + len(ks_orbital_indicies) ) ]
+    """
+
+    # We need to update the indexing of the sd_basis - in terms of the rows and cols of St_KS
+    # reindex ks orbs according to the matrix size
+    n_alp_ks_orbs = len(ks_orbital_indicies)
+    alp_homo_matrix_index = 0
+    for i in range( n_alp_ks_orbs ):
+        if ks_orbital_indicies[i] == ks_orbital_homo_index:
+            alp_homo_matrix_index = i+1
+
+    ks_orbs_new_index = [ i+1 for i in range(n_alp_ks_orbs) ]
+
+    # Form excited state SDs
+    excitations = []
+    # For each Slater determinant basis state, which could have spin_component "alp" or "bet"
+    for j in range( len( sd_basis_states ) ):
+
+        if sd_basis_states[j][1] == "alp":
+            initial_ks_orb = int( sd_basis_states[j][0][0] ) - ks_orbital_homo_index + alp_homo_matrix_index
+            final_ks_orb   = int( sd_basis_states[j][0][1] ) - ks_orbital_homo_index + alp_homo_matrix_index
+            excitations.append( [initial_ks_orb, final_ks_orb]  )
+    
+        elif sd_basis_states[j][1] == "bet":
+            initial_ks_orb = int( sd_basis_states[j][0][0] ) - ks_orbital_homo_index + alp_homo_matrix_index + n_alp_ks_orbs
+            final_ks_orb   = int( sd_basis_states[j][0][1] ) - ks_orbital_homo_index + alp_homo_matrix_index + n_alp_ks_orbs
+            excitations.append( [-initial_ks_orb, -final_ks_orb]  )
+    print( "excitations = ", excitations )
+    #sys.exit(0)
+
+    sd_basis = [ [] ]
+    # We are now going to form the ground state Slater determinant. We first need a loop over all of the Kohn-Sham orbitals
+    # beginning from index 1.
+    # At this point we have a bunch of Slater determinants in the form of reindexed excitations in the variable "excitations" 
+    for i in range( 1, 2*len( ks_orbital_indicies )-1 ):
+        # Form ground state SD. Following the example in the function's documentation, the ground state Slater determinant should be:
+        # [ 1, 2, 3, 4, 5, -11, -12, -13, -14,- 15 ]. Later on, we have the option of using only [ 5, -15 ] when computing the overlaps
+        if i < alp_homo_matrix_index + 1:
+            sd_basis[0].append( i )
+        if i > n_alp_ks_orbs and i < n_alp_ks_orbs + alp_homo_matrix_index + 1:
+            sd_basis[0].append( -i )
+    #print ( sd_basis )
+
+    # Now that we have done the ground state slater 
+    for j in range( len( excitations ) ):
+        sd_excitation = [ excitations[j][1] if x == excitations[j][0] else x for x in sd_basis[0] ]
```

Comment:
no - no - no, don't do all in 1 line - it is very hard to understand; please review in other places of your code to remove such a style